Create selenium tests that capture screenshots of search results

aldenstpage opened the pull request:
Related to creativecommons/cccatalog-api#311.

@kgodey, do you have any suggestions for test search terms?

aldenstpage commented on test/selenium-webdriver/search_screencaps.py:
```diff
@@ -0,0 +1,58 @@
+
+SCREEN_DIR = './screenshots'
+TARGET_URL = 'https://search.creativecommons.org'
+release = rs.get('https://api-dev.creativecommons.engineering/version').json()
```
This is temporarily pointing at api-dev until the next production release of cccatalog-api.

brenoferreira commented on .gitignore:
```diff
@@ -6,6 +6,7 @@ yarn-debug.log*
 yarn-error.log*
 /test/unit/coverage/
 /test/e2e/reports/
+/test/selenium/geckodriver.log
```
I think the `/test/selenium-webdriver/screenshots/*` should be ignored too.

kgodey commented:
Here are suggested search terms in addition to what you have already.

Suggested search terms reported as low quality:
- weather (reported as inappropriate)
- Andrew Jackson (reported as inappropriate)
- maps of the Atlantic
- wildlife
- reading
- Syrian women
- medicine
- health
- periodic table
- aardvark
- children playing
- nurse
- basketball team

Suggested search terms based on frequently searched terms:
- music
- computer
- technology
- car
- books
- video
- beach
- flower
- food
- tree
- people
- nature
- money
- house
- school

aldenstpage commented:
@kgodey I added the search terms you listed and also updated the script to capture the second page of results.

@brenoferreira I spent a few minutes setting up Docker before realizing this isn't a good fit for it. It requires a live Firefox GUI, which means X11 needs to be running. I could probably rig up X11, X forwarding, and remote desktop to completely containerize this, but it would take a few days of work. It may be best to instead follow the README and install the two dependencies manually so you can run it natively (`geckodriver` and `pipenv`)

brenoferreira commented:
@aldenstpage ah, right. Could you instead add a small script that installs the dependencies and gets the tests ready for execution? Also, add an npm script to `package.json` that runs these tests, similar to the unit tests there: https://github.com/creativecommons/cccatalog-frontend/blob/develop/package.json#L11

aldenstpage commented:
@brenoferreira I've added some NPM scripts. Installation of geckodriver and pipenv are platform dependent, so I'm reluctant to include a script for it, but these particular scripts should work on your machine.

brenoferreira commented on package.json:
```diff
@@ -9,6 +9,8 @@
     "dev": "webpack-dev-server --https --inline --progress --config build/webpack-dev-conf.js",
     "start": "npm run dev",
     "unit": "jest --config test/unit/jest-conf.js --coverage",
+    "qa-install": "brew install geckodriver && brew install pipenv",
```
Shouldn't this include `&& cd test/selenium-webdriver && pipenv install`?

aldenstpage replied:
That step is executed with `pipenv run`.

aldenstpage commented:
That step is executed with `pipenv run`

On Mon, Jul 22, 2019, 10:55 AM Breno Ferreira <notifications@github.com>
wrote:

> *@brenoferreira* commented on this pull request.
> ------------------------------
>
> In package.json
> <https://github.com/creativecommons/cccatalog-frontend/pull/417#discussion_r305889097>
> :
>
> > @@ -9,6 +9,8 @@
>      "dev": "webpack-dev-server --https --inline --progress --config build/webpack-dev-conf.js",
>      "start": "npm run dev",
>      "unit": "jest --config test/unit/jest-conf.js --coverage",
> +    "qa-install": "brew install geckodriver && brew install pipenv",
>
> Shouldn't this include && cd test/selenium-webdriver && pipenv install?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/creativecommons/cccatalog-frontend/pull/417?email_source=notifications&email_token=AGEJID6BKT2Q3HSGSSOMLO3QAXC45A5CNFSM4H3WD2EKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB7E2CRI#pullrequestreview-264872261>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGEJID3KVA5C4EIJRSJMCILQAXC45ANCNFSM4H3WD2EA>
> .
>
